crates/optimism/payload/src/payload.rs

```diff
@@ -52,7 +53,11 @@ impl PayloadBuilderAttributes for OptimismPayloadBuilderAttributes {
             .unwrap_or_default()
             .into_iter()
             .map(|data| {
-                TransactionSigned::decode_enveloped(&mut data.as_ref())
+                TransactionSigned::decode_2718(&mut data.as_ref())
```

Comment:
this is needed to avoid introducing eip2718 error variant on `PayloadError`. it can be made nicer once we have https://github.com/alloy-rs/alloy/pull/1359

Reply:
cool, can we track these in a new issue so we don't forget?